Document the list of all instrumented libraries (#286)
signalfx/signalfx-dotnet-tracing: docs/instrumented-libraries.md

# Instumented libraries and framework

## Fully supported

The libraries below should be fully supported and the conventions are inspired by
the [OpenTelemetry Trace Semantic Conventions](https://github.com/open-telemetry/opentelemetry-specification/tree/main/specification/trace/semantic_conventions).

> | ID | Library |

Reviewer: Thanks for capitalizing the product names properly :)

| MongoDb | [MongoDB.Driver.Core](https://www.nuget.org/packages/MongoDB.Driver.Core/) |
| MySql | [MySql.Data](https://www.nuget.org/packages/MySql.Data/) |
| Npgsql | [Npgsql](https://www.nuget.org/packages/Npgsql/) |
| Oracle | [Oracle.ManagedDataAccess](https://www.nuget.org/packages/Oracle.ManagedDataAccess/) |
| RabbitMQ | [RabbitMQ.Client](https://www.nuget.org/packages/RabbitMQ.Client/) |
| ServiceStackRedis | [ServiceStack.Redis](https://www.nuget.org/packages/ServiceStack.Redis/) |
| SqlClient | [Microsoft.Data.SqlClient](https://www.nuget.org/packages/Microsoft.Data.SqlClient/) and [`System.Data.SqlClient`](https://docs.microsoft.com/en-us/dotnet/api/system.data.sqlclient) |
| Sqlite | [SQLite](https://www.nuget.org/packages/SQLite/) |
| StackExchangeRedis | [StackExchange.Redis](https://www.nuget.org/packages/StackExchange.Redis/) |
| Wcf | Windows Communication Foundation (WCF) |
| WebRequest | [`System.Net.WebRequest`](https://docs.microsoft.com/en-us/dotnet/api/system.net.webreques) |
| WinHttpHandler | [`System.Net.Http.WinHttpHandler`](https://docs.microsoft.com/en-us/dotnet/api/system.net.http.winhttphandler) |

## Partially supported

> The libraries below are instrumented, yet the instrumentation, may not

Reviewer: ```suggestion
The libraries below are instrumented, though the instrumentation might not
```

produce appropriate spans.

| ID | Library |
| -- | ---     |
| AwsSdk | [AWSSDK.Core](https://www.nuget.org/packages/AWSSDK.Core/) |
| AwsSqs | [AWSSDK.SQS](https://www.nuget.org/packages/AWSSDK.SQS/) |
| AzureFunctions | [Microsoft.Azure.WebJobs](https://www.nuget.org/packages/Microsoft.Azure.WebJobs/) |
| CosmosDb | [Microsoft.Azure.Cosmos](https://www.nuget.org/packages/Microsoft.Azure.Cosmos/) |
| Couchbase | [CouchbaseNetClient](https://www.nuget.org/packages/CouchbaseNetClient/) |
| Msmq | [`System.Messaging`](https://docs.microsoft.com/en-us/dotnet/api/system.messaging) |
| MsTestV2 | [Microsoft.VisualStudio.TestPlatform](https://www.nuget.org/packages/Microsoft.VisualStudio.TestPlatform/) |
| NUnit | [NUnit](https://www.nuget.org/packages/NUnit/) |
| ServiceRemoting | [Microsoft.ServiceFabric.Services.Remoting](https://www.nuget.org/packages/Microsoft.ServiceFabric.Services.Remoting/) |
| XUnit | [xunit](https://www.nuget.org/packages/xunit) |

## Details

> You can get exact information on the instrumented code [here](../tracer/src/Datadog.Trace/ClrProfiler/InstrumentationDefinitions.Generated.cs).

Reviewer: What does "exact" mean in this context? How should this information be used?

Author: What methods from what assemblies are being instrumented.

It is useful if the customer would like to check e.g. the version range of the library that we instrument. It can be used also to let know what exact API  is being instrumented.

Any idea how to describe it better?

Reviewer: ```suggestion
You can find out which methods and assemblies are being instrumented by inspecting the [InstrumentationDefinitions](../tracer/src/Datadog.Trace/ClrProfiler/InstrumentationDefinitions.Generated.cs) file. This is useful to check, for example, the version range of the library you're instrumenting.
```
